file/builder.go

```diff
@@ -882,6 +918,37 @@ func (b *stateBuilder) plugins() {
 			}
 			p.Route = utils.GetRouteReference(r.Route)
 		}
+		if p.ConsumerGroup != nil && !utils.Empty(p.ConsumerGroup.ID) {
+			cg, err := b.intermediate.ConsumerGroups.Get(*p.ConsumerGroup.ID)
+			if errors.Is(err, state.ErrNotFound) {
+				b.err = fmt.Errorf("consumer-group %v for plugin %v: %w",
+					p.ConsumerGroup.FriendlyName(), *p.Name, err)
+				return
+			} else if err != nil {
+				b.err = err
+				return
+			}
+			p.ConsumerGroup = utils.GetConsumerGroupReference(cg.ConsumerGroup)
+		}
+
+		if (b.isConsumerGroupScopedPluginSupported && !b.isKonnect) && *p.Name == ratelimitingAdvancedPluginName {
+			// check if deprecated consumer-groups configuration is present in the config
+			var consumerGroupsFound bool
+			if groups, ok := p.Config["consumer_groups"]; ok {
+				// if groups is an array of length > 0, then consumer_groups is set
+				if groupsArray, ok := groups.([]interface{}); ok && len(groupsArray) > 0 {
+					consumerGroupsFound = true
+				}
+			}
+			_, enforceConsumerGroupsFound := p.Config["enforce_consumer_groups"]
+			if consumerGroupsFound || enforceConsumerGroupsFound {
+				b.err = errors.New("a rate-limiting-advanced plugin with config.consumer_groups\n" +
+					"and/or config.enforce_consumer_groups was found. Please use Consumer Groups scoped\n" +
+					"Plugins when running against Kong Enterprise 3.4.0 and above.\n\n" +
+					"Check DOC_LINK for more information")
+				return
+			}
+		}
```

Comment:
How about a separate function for this fragment (to make it look on the call site e.g. `err := b.validatePlugin(p)`?